gateway/app/main/processes/controller.py

```diff
@@ -1,11 +1,65 @@
 from .federated_learning_process import FLProcess
+from .federated_learning_cycle import FederatedLearningCycle
 
 
 class FLPController:
-    """ This class implements controller design pattern over the federated learning processes."""
+    """ This class implements controller design pattern over the federated learning processes. """
 
     def __init__(self):
         self.processes = {}
+        self._workers = {}
+
+    def create_cycle(self, model_id: str, worker_id: str, cycle_time: int = 2500):
```

Comment:
should we add a attribute maybe `cycle_sequence` to track:
- for each specific instance of the cycle 
- being to a specific (model_id, model_version) tuple
- what is the sequence number?

The reason being...  in https://github.com/OpenMined/Roadmap/blob/master/web_and_mobile_team/projects/federated_learning.md
under 2. test... we have a server config that has a field called `do_not_reuse_workers_until_cycle`
and so we need a way to track and tell if we should reject a worker who requested to join (model_123, version_2)
at the 5th cycle after participating in cycle=1... we should reject if `do_not_reuse_workers_until_cycle = 5`, we should not reject base on this if `do_not_reuse_workers_until_cycle = 3`
(yet another reason we gona need to stash this info into a db : )

Reply:
Done! :)

Reply:
Correct, good find @Prtfw. Resolving!